### PR TITLE
MSC2451: Remove `query_auth` federation endpoint.

### PR DESCRIPTION
**Pull request by clokep**
[Rendered](https://github.com/matrix-org/matrix-doc/blob/clokep/remove-old-fed-endpoint/proposals/2451-remove-query_auth-federation-endpoint.md).

This is the first MSC I've put up, so let me know if I've done anything very wrong!

### CODE REVIEW

File: proposals/2451-remove-query_auth-federation-endpoint.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# MSC2451: Remove the `query_auth` federation endpoint
+
+The `query_auth` federation endpoint is unused by Synapse and should be removed.
```
Comment (turt2live) [inline proposals/2451-remove-query_auth-federation-endpoint.md]: There's other server implementations than Synapse, so at the spec level the justification isn't whether or not Synapse cares but whether or not the endpoint serves a purpose and if that purpose is worthwhile.

Comment (clokep) [reply proposals/2451-remove-query_auth-federation-endpoint.md]: Yes, I've checked other implementations and it is unused in all.

Hopefully @richvdh can help fill in some of the reasoning behind why it is no longer worthwhile.

File: proposals/2451-remove-query_auth-federation-endpoint.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+that point this endpoint is not called).
+
+During removal it was noted that the code to call this endpoint was already
+unreachable. It seems that this code was never reachable and was meant for an
+error situation which was never built out (see `git log -S NOT_ANCESTOR`, the
+error condition is never assigned).
+
+## Alternatives
+
+The endpoint could be deprecated in removed in a future version of the specification.
```
Comment (ara4n) [inline proposals/2451-remove-query_auth-federation-endpoint.md]:
```suggestion
The endpoint could be deprecated and removed in a future version of the specification.
```
i think?

File: proposals/2451-remove-query_auth-federation-endpoint.md (hunk 1 above)
Comment (clokep) [reply proposals/2451-remove-query_auth-federation-endpoint.md]: From some conversation on Matrix it seems like the reasoning is (stealing from @ara4n):

> This API was added without sufficient thought nor testing. The endpoint isn't called in any known implementations, and we do not believe it to be necessary for federation to work. The only known implementation (in Synapse) was not fully fleshed out and is broken.

Comment (turt2live) [reply proposals/2451-remove-query_auth-federation-endpoint.md]: That'll need to make it into the MSC, as the MSC really shouldn't be referencing Synapse directly. Though Synapse's brokenness still isn't a reason to remove it - the endpoint's uselessness is.

Comment (clokep) [reply proposals/2451-remove-query_auth-federation-endpoint.md]: @turt2live Yes, I figured having a discussion in the comments and coming to a conclusion first before pushing more commits made sense. Please let me know if this isn't how the process usually works.

Comment (turt2live) [reply proposals/2451-remove-query_auth-federation-endpoint.md]: Acknowledgement of the concern is usually enough, just need the edits made before things can enter FCP is all.

File: proposals/2451-remove-query_auth-federation-endpoint.md (hunk 2 above)
Comment (clokep) [reply proposals/2451-remove-query_auth-federation-endpoint.md]: Yep. 👍

File: proposals/2451-remove-query_auth-federation-endpoint.md (hunk 1 above)
Comment (clokep) [reply proposals/2451-remove-query_auth-federation-endpoint.md]: I updated this and included some additional information provided by @erikjohnston

File: proposals/2451-remove-query_auth-federation-endpoint.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+## Proposal
+
+Remove the following endpoint:
+
+* [POST `/_matrix/federation/v1/query_auth/{roomId}/{eventId}`](https://matrix.org/docs/spec/server_server/r0.1.3#post-matrix-federation-v1-query-auth-roomid-eventid)
+
+## Potential issues
+
+Removing this endpoint impacts backwards compatibility, in practice removing
```
Comment (richvdh) [inline proposals/2451-remove-query_auth-federation-endpoint.md]: I'm not sure it's fair to say it impacts backwards compatibility, given the synapse code was unreachable even before 1.5?

**Comment by richvdh** [comment]
@mscbot fcp merge

File: proposals/2451-remove-query_auth-federation-endpoint.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+# MSC2451: Remove the `query_auth` federation endpoint
+
+This API was added without sufficient thought nor testing. The endpoint isn't
+used in any known implementations, and we do not believe it to be necessary
+for federation to work. The only known implementation (in Synapse) was not fully
+fleshed out and is broken.
+
+For background, the idea behind this endpoint was that two homeservers would be
+able to share state events with the hope of filling in missing state from one
```
Comment (ara4n) [inline proposals/2451-remove-query_auth-federation-endpoint.md]: state events or auth events?

**Comment by ara4n** [comment]
if @mscbot was working, i would check its tickbox to merge.

**Comment by turt2live** [comment]
Implementation was at https://github.com/matrix-org/synapse/pull/7026 ftr

**Comment by turt2live** [comment]
Merged 🎉

**Pull request merged**